modules/compute/eks/stack.py

```diff
@@ -359,6 +376,9 @@ def __init__(
                                 ),
                             )
                         ],
+                        metadata_options=ec2.CfnLaunchTemplate.MetadataOptionsProperty(
+                            http_tokens="required", http_put_response_hop_limit=2
```

Comment:
what are the ramifications of this limit (comparing public vs. private subnets) if any??

Reply:
In general, IMDSv2 is recommended because it is more secure. DCV clients are using IMDSv2. If we disable IMDSv2, DCV will stuck and does not work.

When we enable IMDSv2, there is another requirement to set http_put_response_hop_limit, which specifies how many hops are allowed for ec2 instance to reach IMDS in the case of EKS. For the traffic to be reached from the Pod, it requires the network to hop from containerized applications in EKS that are running in separate network namespaces from the EC2 instance. Therefore, a higher limit (2 or higher) is required. The default value is 1 in launch template.

Some posts say that it is 2 as default (https://aws.amazon.com/about-aws/whats-new/2020/08/amazon-eks-supports-ec2-instance-metadata-service-v2/), but I was still experiencing networking issues when we left the value default. It would be better to clearly specify the value to be 2.

Reply:
Nice